src/applications/disability-benefits/all-claims/config/submitForm.js

```diff
@@ -3,66 +3,69 @@ import {
   recordEvent,
 } from 'us-forms-system/lib/js/helpers';
 
-export default function submitForm(form, formConfig) {
-  const body = formConfig.transformForSubmit
-    ? formConfig.transformForSubmit(formConfig, form)
-    : transformForSubmit(formConfig, form);
+const submitFormFor = eventName =>
+  function submitForm(form, formConfig) {
+    const body = formConfig.transformForSubmit
+      ? formConfig.transformForSubmit(formConfig, form)
+      : transformForSubmit(formConfig, form);
 
-  // Copied and pasted from USFS with a couple changes:
-  // 1. Sets `withCredentials` to true
-  // 2. Sends the Authorization header with the user token
-  return new Promise((resolve, reject) => {
-    const req = new XMLHttpRequest();
-    req.open('POST', formConfig.submitUrl);
+    // Copied and pasted from USFS with a couple changes:
+    // 1. Sets `withCredentials` to true
+    // 2. Sends the Authorization header with the user token
+    return new Promise((resolve, reject) => {
+      const req = new XMLHttpRequest();
+      req.open('POST', formConfig.submitUrl);
 
-    req.withCredentials = true;
+      req.withCredentials = true;
 
-    req.addEventListener('load', () => {
-      if (req.status >= 200 && req.status < 300) {
-        recordEvent({ event: 'form-submit-successful' });
+      req.addEventListener('load', () => {
+        if (req.status >= 200 && req.status < 300) {
+          recordEvent({ event: `${eventName}--submission-successful` });
```

Comment:
This is the only actual change to `submitForm` itself. I made this a partially applied function for added flexibility, but that unfortunately meant a lot of added whitespace which artificially inflates the diff line count.